### PR TITLE
[CROSSDATA-399] Refactor connection management in native execution for MongoDB

### PR DESCRIPTION
**Pull request by darroyocazorla**

### CODE REVIEW

**Comment by miguel0afd** [comment]
Btw, could you, please, include this new command in the documentation: https://stratio.atlassian.net/wiki/display/CROSSDATA/Reference+Guide?

**Comment by miguel0afd** [comment]
+1

**Comment by miguel0afd** [comment]
LGTM

File: mongodb/src/main/scala/com/stratio/crossdata/connector/mongodb/DefaultSource.scala [hunk 1]
```diff
@@ -167,20 +166,68 @@ class DefaultSource extends ProviderDS with TableInventory with DataSourceRegist
...
+  def parseParametersWithoutValidation(parameters : Map[String,String]): Config = {
+
+    // required properties
+    /** We will assume hosts are provided like 'host:port,host2:port2,...' */
+    val properties: Map[String, Any] = parameters.updated(Host, parameters.getOrElse(Host, notFound[String](Host)).split(",").toList)
+
+    //optional parseable properties
+    val optionalProperties: List[String] = List(Credentials,SSLOptions, UpdateFields)
+
+    val finalProperties = (properties /: optionalProperties){
```
Comment (pfcoperez) [inline mongodb/src/main/scala/com/stratio/crossdata/connector/mongodb/DefaultSource.scala]: As beautiful as it gets

File: mongodb/src/main/scala/com/stratio/crossdata/connector/mongodb/DefaultSource.scala [hunk 2]
```diff
@@ -167,20 +166,68 @@ class DefaultSource extends ProviderDS with TableInventory with DataSourceRegist
...
+      /** We will assume credentials are provided like 'user,database,password;user,database,password;...' */
+      case (properties,Credentials) =>
+        parameters.get(Credentials).map{ credentialInput =>
+          val credentials = credentialInput.split(";").map(_.split(",")).toList
+            .map(credentials => MongodbCredentials(credentials(0), credentials(1), credentials(2).toCharArray))
```
Comment (pfcoperez) [inline mongodb/src/main/scala/com/stratio/crossdata/connector/mongodb/DefaultSource.scala]: Just an aestetic & subjective suggestion:

```
val credentials = credentialInput.split(";").map(_.split(",").toList).map {
  case a::b::c::Nil => MongodbCredentials(a, b, c.toCharArray))
}
```

**Comment by pfcoperez** [comment]
+1

**Pull request merged**